src/os/unix.rs

```diff
@@ -142,8 +153,22 @@ impl WindowExt for Window {
 
 /// Additional methods on `WindowBuilder` that are specific to Unix.
 pub trait WindowBuilderExt {
-
+    fn with_x11_visual<T>(self, visual_infos: *const T) -> WindowBuilder;
+    fn with_x11_screen(self, screen_id: i32) -> WindowBuilder;
 }
 
 impl WindowBuilderExt for WindowBuilder {
+    #[inline]
+    fn with_x11_visual<T>(mut self, visual_infos: *const T) -> WindowBuilder {
+        self.platform_specific.visual_infos = Some(
+            unsafe { ptr::read(visual_infos as *const XVisualInfo) }
```

Comment:
Shouldn't `with_x11_visual` be `unsafe`? I can't find any issue open about this, and I'm not sure about backwards compatibility but this seems unsound as-is.